torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -287,9 +287,97 @@ void removeNopPacking(std::shared_ptr<Graph>& graph) {
 
     n->removeAllInputs();
     it.destroyCurrent();
+
+    if (input->outputs()[0]->uses().size() == 0) {
+      input->destroy();
+    }
+  }
+}
+
+void fixDefaultRNNState(std::shared_ptr<Graph>& graph, Node * n, int input_index) {
+  auto initial_state = n->inputs()[input_index];
+  if (initial_state->node()->kind() == onnx::Constant ||
+      (initial_state->node()->kind() == onnx::Slice &&
+       initial_state->node()->inputs()[0]->node()->kind() == onnx::Constant)) {
+    Node * shape_of_input = graph->create(onnx::Shape, 1);
+    shape_of_input->insertBefore(n);
+    shape_of_input->addInput(n->inputs()[0]);
+
+    Node * gather_indices = graph->create(onnx::Constant, 1);
+    gather_indices->insertBefore(n);
+    gather_indices->t_(attr::value, at::Scalar(1).toTensor());
```

Comment:
What's the expected shape of the input for RNNs? Is batch dimension `1` and not `0`?

Reply:
yes batch dimension is always 1. even with batch_first=True because a transpose is inserted earlier on. Would you like me to link to the docs here? (shape is described at http://pytorch.org/docs/master/nn.html#torch.nn.RNN)